src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -288,22 +293,52 @@ public function walkSelectorSource(QOM\SelectorInterface $source)
     }
 
     /**
+     * @param QOM\JoinConditionInterface $right
+     *
+     * @return string the alias on the right side of a join
+     *
+     * @throws \BadMethodCallException if the provided JoinCondition has no valid way of getting the right selector
+     */
+    private function getRightJoinSelector(QOM\JoinConditionInterface $right)
+    {
+        if ($right instanceof QOM\ChildNodeJoinConditionInterface) {
+            return $right->getParentSelectorName();
+        } elseif ($right instanceof QOM\DescendantNodeJoinConditionInterface) {
+            return $right->getAncestorSelectorName();
+        } elseif (method_exists($right, 'getSelector2Name')) {
+            return $right->getSelector2Name();
```

Comment:
I've changed this in my fork https://github.com/richard-fizz/jackalope-doctrine-dbal
Is there a way I can push this to this PR?